YatimaStdLib/RBMap.lean

```diff
@@ -2,20 +2,35 @@ import Lean
 
 namespace Std.RBMap
 
-variable {cmp : α → α → Ordering} 
+variable {cmp : α → α → Ordering}
 
-instance : Inhabited (RBMap α β cmp) where
-  default := .empty
-
-def single (a : α) (b : β) : RBMap α β cmp := 
+def single (a : α) (b : β) : RBMap α β cmp :=
   RBMap.empty.insert a b
 
-def enumList (xs : List α) : RBMap α Nat cmp := 
+def enumList (xs : List α) : RBMap α Nat cmp :=
   RBMap.ofList $ xs.enum.map (fun (x, y) => (y, x))
 
+def unitMap (xs : List α) : RBMap α Unit cmp :=
+  RBMap.ofList $ xs.map (fun x => (x, ()))
+
 def filterOut [BEq α] [Ord α] (map : RBMap α β cmp) (t : RBTree α cmp) :
     RBMap α β cmp :=
   map.fold (init := default) fun acc a b =>
     if t.contains a then acc else acc.insert a b
 
+def toKeyList : RBMap α β cmp → List α
+  | ⟨t, _⟩ => t.revFold (fun ks k _ => k::ks) []
+
+def toValueList : RBMap α β cmp → List β
+  | ⟨t, _⟩ => t.revFold (fun vs _ v => v::vs) []
```

Comment:
Ideas:
* Call these `keys` and `values` for brevity
* Use an `Array` as an intermediary data structure and use `Array.push` for fast appending. Then return the array itself in the end instead of a list, since arrays have broader usages for optimized indexings. This will keep the original order defined by the `Ord` instance